src/main/scala/com/exasol/cloudetl/kinesis/KinesisUserProperties.scala

```diff
@@ -65,6 +70,32 @@ class KinesisUserProperties(val propertiesMap: Map[String, String])
   final def getTableName(): String =
     getString(TABLE_NAME_PROPERTY)
 
+  final def containsMaxRecordsPerRun(): Boolean =
+    containsKey(MAX_RECORDS_PER_RUN_PROPERTY)
+
+  final def getMaxRecordsPerRun(): String =
+    getString(MAX_RECORDS_PER_RUN_PROPERTY)
+
+  /**
+   * Returns a new [[KinesisUserProperties]] that merges the key-value pairs
+   * parsed from user provided Exasol named connection object.
+   */
+  final def mergeWithConnectionObject(exaMetadata: ExaMetadata): KinesisUserProperties = {
+    validateConnectionObject()
+    val connectionParsedMap =
+      parseConnectionInfo(AWS_ACCESS_KEY_PROPERTY, Option.apply(exaMetadata))
```

Comment:
```suggestion
      parseConnectionInfo(AWS_ACCESS_KEY_PROPERTY, Option(exaMetadata))
```

Reply:
With `apply` methods you can remove calling it specifically.

Reply:
done